nekoyume/Assets/_Scripts/UI/Widget/System/HasIconAndButtonSystem.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections;
+using Nekoyume.Game.Controller;
+using Nekoyume.L10n;
+using Nekoyume.UI.Module;
+using TMPro;
+using UnityEngine;
+
+namespace Nekoyume.UI
+{
+    public class HasIconAndButtonSystem : SystemWidget
+    {
+        public enum SystemType : int
+        {
+            Error = 0,
+            BlockChainError,
+            Information,
+        }
+
+        [SerializeField]
+        private IconAndButton[] uiBySystemType;
+
+        [SerializeField]
+        private Blur blur;
+
+        private TextButton _confirmButton = null;
+
+        private TextButton _cancelButton = null;
+
+        private TextMeshProUGUI _contentText = null;
+
+        private TextMeshProUGUI _titleText = null;
+
+        public System.Action ConfirmCallback { get; set; }
+        public System.Action CancelCallback { get; set; }
+
+        protected override void Awake()
+        {
+            base.Awake();
+
+            SubmitWidget = Confirm;
+            CloseWidget = Cancel;
+            foreach (var ui in uiBySystemType)
+            {
+                ui.confirmButton.OnClick = Confirm;
+                ui.cancelButton.OnClick = Cancel;
+            }
+        }
+
+        public override void Close(bool ignoreCloseAnimation = false)
+        {
+            foreach (var ui in uiBySystemType)
+            {
+                ui.gameObject.SetActive(false);
+            }
+            if (blur)
+            {
+                blur.Close();
+            }
+            base.Close(ignoreCloseAnimation);
+        }
+
+        public void Show(string title, string content, string labelYes = "UI_OK",
+            bool localize = true, SystemType type = SystemType.Error)
+        {
+            if (blur)
+            {
+                blur.Show();
+            }
+
+            if (gameObject.activeSelf)
+            {
+                Close(true);
+                ShowWithTwoButton(title, content, labelYes, "", localize, false, type);
+                return;
+            }
+
+            Set(title, content, labelYes, "", localize, false, type);
+        }
+
+        public void ShowWithTwoButton(string title, string content, string labelYes = "UI_OK", string labelNo = "UI_CANCEL",
+            bool localize = true, bool hasConfirmButton = true, SystemType type = SystemType.Error)
+        {
+            if (blur)
+            {
+                blur.Show();
+            }
+
+            if (gameObject.activeSelf)
+            {
+                Close(true);
+                ShowWithTwoButton(title, content, labelYes, labelNo, localize, hasConfirmButton, type);
+                return;
+            }
+
+            Set(title, content, labelYes, labelNo, localize, hasConfirmButton, type);
+            Show();
+        }
```

Comment:
I think this class has too many roles like `ShowByXXX`. Can they be separated? (like using inheritance or separating class)

Reply:
It's not impossible, but it's a hassle to do now. I will merge it first and improve it later.

Reply:
Thanks to your opinion. I will make sure to separate it as soon as possible!